aws_xray_sdk/core/recorder.py

```diff
@@ -296,8 +298,11 @@ def begin_subsegment(self, name, namespace='local'):
             log.warning("No segment found, cannot begin subsegment %s." % name)
             return None
 
-        if not segment.sampled:
+        current_entity = self.get_trace_entity()
+
+        if not current_entity.sampled or not sampling:
```

Comment:
Would looking at just current_entity.sampled be enough here?

Reply:
I don't think so. If the current entity (segment or subsegment) is sampled and a new unsampled subsegment is being added, it is necessary to check the sampling flag being set on the newly created subsegment.